hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsOutputStream.java

```diff
@@ -108,23 +106,64 @@ public AbfsOutputStream(
             .isDisableOutputStreamFlush();
     this.lastError = null;
     this.lastFlushOffset = 0;
-    this.bufferSize = abfsOutputStreamContext.getWriteBufferSize();
-    this.buffer = byteBufferPool.getBuffer(false, bufferSize).array();
     this.bufferIndex = 0;
-    this.writeOperations = new ConcurrentLinkedDeque<>();
     this.outputStreamStatistics = abfsOutputStreamContext.getStreamStatistics();
-
-    this.maxConcurrentRequestCount = 4 * Runtime.getRuntime().availableProcessors();
-
-    this.threadExecutor
-        = new ThreadPoolExecutor(maxConcurrentRequestCount,
-        maxConcurrentRequestCount,
-        10L,
-        TimeUnit.SECONDS,
-        new LinkedBlockingQueue<>());
-    this.completionService = new ExecutorCompletionService<>(this.threadExecutor);
     this.cachedSasToken = new CachedSASToken(
         abfsOutputStreamContext.getSasTokenRenewPeriodForStreamsInSeconds());
+
+    init(abfsOutputStreamContext);
+    buffer = new byte[bufferSize];
+  }
+
+  private void init(final AbfsOutputStreamContext abfsOutputStreamContext) {
+    if (isCommonPoolsInitialised()) {
+      return;
+    }
+
+    initWriteBufferPool(abfsOutputStreamContext);
+
+    ThreadFactory daemonThreadFactory = new ThreadFactory() {
```

Comment:
there's some hadoop thread factory you should be able to lift. Also: name the threads